lib/jekyll/assets/env.rb

```diff
@@ -142,9 +148,9 @@ def liquid_proxies
       # @return [Pathname/Pathutil]
       # --
       def in_cache_dir(*paths)
-        jekyll.in_source_dir(asset_config["cache"] || ".asset-cache",
-          *paths
-        )
+        paths.reduce(@cache_path) do |base, path|
+          Jekyll.sanitized_path(base, path)
+        end
```

Comment:
Mimics `Jekyll::Site#in_source_dir` implementation. Resolves path relative to the cache path, which is no longer required to live within the source dir.